### PR TITLE
fix(filter): Allow generic Slackbot to pass

### LINKED ISSUE #14814: Legit errors from Slack commands were filtered out
## Important Details

How are you running Sentry?

* [ ] On-Premise docker [Version xyz]
* [X] Saas (sentry.io)
* [ ] Other [briefly describe your environment]

## Description

We accept Slack unfurling and interactive commands. Those requests from Slack server may lead to errors we do want to know and track. However, Sentry server broadly rejects those errors due to Slack's user agent (done in #5284). There is no option to only whitelist slack bots and we would like to avoid turning off the crawler filtering as a whole.

Note Slack now has [muiltiple bot agent](https://api.slack.com/robots) so one possibility is only to block the crawler ones.

### Possible Solution

- Remove slack filter
- Only filter slack crawler agents but allow slackbot
- Allow us to turn off only the slack rule

### PR DESCRIPTION
**Pull request by BYK**
Fixes getsentry/sentry#14814

### CODE REVIEW

**Comment by BYK** [comment]
Eh, turns out this is not as straightforward as I thought. The generic `bots?` term eliminates the legit `Slackbot 1.0` agent and the RegEx crate we use don't have lookaround support to only match when `bot` is _not_ preceded with `Slack`.

The only two options I see are:
1. Remove the generic term (this has been there for 5+ years, scary)
2. Add a second layer that uses an allowlist and add Slackbot 1.0 there for by-pass

@jan-auer @untitaker thoughts?

**Comment by jan-auer** [comment]
Also adding @lobsterkatie since she's been adding filters over time. I have no context on implications of these filters.

For context, this was originally added due to https://github.com/getsentry/sentry/issues/5284 in https://github.com/getsentry/sentry/pull/5285

**Comment by jan-auer** [comment]
@BYK could you add more test cases for the bots we still want to block?

**Comment by BYK** [comment]
>@BYK could you add more test cases for the bots we still want to block?

They are already there, that's why I didn't touch the tests much.

**Comment by lobsterkatie** [comment]
Happy to help but I have no special context on this. Over time in support, people would write in, asking for things to be added. I'd post in the engineering channel and get someone to give it a thumbs-up, and then make a PR adding to the file. It wasn't a super common thing, though.

**Comment by lobsterkatie** [comment]
Happy to help but I have no special context on this. Over time in support, people would write in, asking for things to be added. I'd post in the engineering channel and get someone to give it a thumbs-up, and then make a PR adding to the file. It wasn't a super common thing, though.

**Comment by untitaker** [comment]
Wouldn't it be possible to enumerate linkexpanding and imgproxy instead?

**Comment by untitaker** [comment]
Answer: We have a generic bots filter that would match slackbot anyway.

File: relay-filter/src/web_crawlers.rs [hunk 1]
```diff
@@ -11,7 +11,7 @@ use crate::{FilterConfig, FilterStatKey};
 /// Checks if the event originates from a known web crawler.
 pub fn matches(event: &Event) -> bool {
     if let Some(user_agent) = user_agent::get_user_agent(event) {
-        WEB_CRAWLERS.is_match(user_agent)
+        !ALLOWED_WEB_CRAWLERS.is_match(user_agent) && WEB_CRAWLERS.is_match(user_agent)
```
Comment (untitaker) [inline relay-filter/src/web_crawlers.rs]: Let's swap this conditional such that for the majority of UAs, only one regex is executed.

**Review by untitaker — CHANGES_REQUESTED** [review]
I think we won't get around having two regexes, but one last comment before we merge.

File: relay-filter/src/web_crawlers.rs (hunk 1 above)
Comment (BYK) [reply relay-filter/src/web_crawlers.rs]: Done :)

**Review by untitaker — APPROVED** [review]

**Pull request merged**